lib/fast_jsonapi/object_serializer.rb

```diff
@@ -51,9 +52,10 @@ def hash_for_collection
 
       data = []
       included = []
+      fieldset = @fieldsets[self.class.reflected_record_type.to_sym]
```

Comment:
`reflected_record_type` is the meaningful default it can be overridden in `set_type` so it probably makes sense here to use `self.record_type`

Reply:
👍

Reply:
@shishirmk Seems like `record_type` is not being set for inherited serializers. Got this when I introduced a spec for it:

```
  1) FastJsonapi::ObjectSerializer sets the correct record type
     Failure/Error: expect(EmployeeSerializer.record_type).to eq 'employee'

       expected: "employee"
            got: nil

       (compared using eql?)
     # ./spec/lib/object_serializer_inheritance_spec.rb:99:in `block (2 levels) in <top (required)>'
```

I'll write a fix for that as well.

Reply:
@shishirmk Looks like `record_type` is not being set for inherited serializers. I added a spec for it and got this:

```
  1) FastJsonapi::ObjectSerializer sets the correct record type
     Failure/Error: expect(EmployeeSerializer.record_type).to eq 'employee'

       expected: "employee"
            got: nil

       (compared using ==)
     # ./spec/lib/object_serializer_inheritance_spec.rb:99:in `block (2 levels) in <top (required)>'
```

Will write a fix for that as well.

Reply:
Yup go ahead and fix that as well.